daemon/daemon.go

```diff
@@ -250,15 +250,9 @@ func (d *Daemon) Dying() <-chan struct{} {
 	return d.tomb.Dying()
 }
 
-func (d *Daemon) auther(r *http.Request) (store.Authenticator, error) {
+func (d *Daemon) auther(user *auth.UserState) (store.Authenticator, error) {
```

Comment:
Yes, this really doesn't feel like the right way to go about it. We literally just looked up this user, building it out of the state, and went over the trouble of validating it. We might have cached information in there or whatever else, and we're now throwing it off to rebuild it out of the state once more via tis id.

There's also more to it than that, which I was hoping to discuss with you in that interactive conversation I suggested a couple of weeks ago. The most important reason we moved over is because the failure modes become awkward if we don't have the user at hand. We may potentially be in the middle of an operation when then user information goes away, and we might have to through a 403 to something that was already partially acted upon.

Compare these two implementations for example:

```
        auther, err := c.d.auther(r)
        if err != nil && err != auth.ErrInvalidAuth {
                return InternalError("%v", err)
        }
```

vs.

```
        if user != nil {
                auther = user.Authenticator()
        }
```

To me it's pretty clear which one is more error prone and will very likely create nasty bugs soon.

Now, what I was hoping to discuss with you is that none of that seems necessary. We can very easily carry around some sort of device-specific information in an auth.User. It makes some sense.. it's the user of a device. Also, note how we have a `MacaroonAuthenticator` being returned from user.Authenticator. If a device auth is also a macaroon, it also sounds fine to have that same authenticator know how to populate a device authenticator.

That said, we didn't really discuss much how these preliminary ideas fit into the whole, so I'm not sure about how much of that is spot on or completely off.